saleor/graphql/account/mutations/permission_group.py

```diff
@@ -36,19 +38,54 @@ class Meta:
         description = "Create new permission group."
         model = auth_models.Group
         permissions = (AccountPermissions.MANAGE_STAFF,)
-        error_type_class = AccountError
-        error_type_field = "account_errors"
+        error_type_class = BulkAccountError
+        error_type_field = "bulk_account_errors"
 
     @classmethod
     def clean_input(cls, info, instance, data):
         cleaned_input = super().clean_input(info, instance, data)
+        errors = defaultdict(list)
         # clean and prepare permissions
         if "permissions" in cleaned_input:
+            indexes = get_permissions_user_has_not(
+                info.context.user, cleaned_input["permissions"]
+            )
+            if indexes:
+                error_msg = "You can't add permission that you don't have."
+                cls.update_errors(
+                    errors,
+                    error_msg,
+                    "permissions",
+                    AccountErrorCode.NO_PERMISSION,
+                    indexes,
+                )
             cleaned_input["permissions"] = get_permissions(cleaned_input["permissions"])
+        if errors:
+            raise ValidationError(errors)
         return cleaned_input
 
+    @classmethod
+    def update_errors(cls, errors, msg, field, code, indexes):
+        for index in indexes:
+            error = ValidationError(msg, code=code, params={"index": index})
+            errors[field].append(error)
 
-class PermissionGroupInput(graphene.InputObjectType):
+    @classmethod
+    def handle_typed_errors(cls, errors: list, **extra):
```

Comment:
This function seems duplicated in few other mutations currently. I would instead try to refactor the original `BaseMutation.handle_typed_errors` and see if we can put this logic there. Then, `BaseMutation` could work both with error types that support indexes as well as with those, that don't. We could then also drop duplicated occurrences of this function in `bulk_mutations/products.py`.

Reply:
I would at least check if we can do that, if not then maybe it has to stay that way for now. But it will have to be refactored anyway at some point.